### ISSUE TITLE
'use before declaration' error is not reported for block scoped variables in case block

### ISSUE THREAD

**Issue by vladima** [Microsoft/TypeScript#5174]
```ts
declare function use(x: any);

function f(x) {
    switch(x) {
        case 1:
            let x = 1;
        case 2:
            x = 2; // should be error because of TDZ
    }
}
```

probably lo-pri, should be fixed if / when we add definite assignment check

**Comment by mhegazy** [comment]
subsumed by #5207
